### PR TITLE
Fix uvfits coords

### PR DESCRIPTION
**Pull request by plaplant**
This PR fixes how antenna coordinates are saved in the model visibility files generated by omnical. Proper coordinate handling requires a coordinate transformation, a change in units, and making antenna positions relative to the telescope location. Note that for instances where a calfile is used, rather than the information inside of the file itself, the antenna positions are not guaranteed to be correct. However, antenna positions were not correct before, and so there is no net change.

### CODE REVIEW

File: hera_cal/omni.py [hunk 1]
```diff
@@ -823,16 +825,59 @@ def make_uvdata_vis(aa, m, v, xtalk=False):
...
+
+    # Save antenna information.
+    if len(ants_telescope) < uv.Nants_data:
+        # Not enough valid antenna positions; default to antennas in data
+        # and give them positions of 0.
+        uv.Nants_telescope = uv.Nants_data
+        uv.antenna_numbers = np.asarray(sorted(ants_data))
+        uv.antenna_positions = np.zeros((uv.Nants_data, 3))
+    else:
+        # Save antenna positions we computed.
```
Comment (zakiali) [inline hera_cal/omni.py]: So for the current way we run things we'll have 143 (I think that's the largest antenna number so far?) antenna positions, antenna numbers, and Nants_telescope = 143?

File: hera_cal/omni.py [hunk 2]
```diff
@@ -823,16 +825,59 @@ def make_uvdata_vis(aa, m, v, xtalk=False):
...
+            antpos[idx, :] = rel_pos
+
+            # also save antenna number to list of antennas
+            ants_telescope.append(iant)
+
+            # increment counter
+            idx += 1
+
+    # Save antenna information.
+    if len(ants_telescope) < uv.Nants_data:
```
Comment (zakiali) [inline hera_cal/omni.py]: Would we ever run into this situation with the way things are currently run?

File: hera_cal/omni.py (hunk 1 above)
Comment (plaplant) [reply hera_cal/omni.py]: Not quite. pyuvdata doesn't require you to have 1..ant_max antennas in the array. Nants_telescope for current data is 47, and there are 47 antenna numbers/positions. The mapping between antenna number and position is given by enumerating uv.antenna_numbers.

File: hera_cal/omni.py (hunk 2 above)
Comment (plaplant) [reply hera_cal/omni.py]: We shouldn't. I left this here primarily for backwards compatibility with previous tests, but it also ensures we produce sensible UVData objects where Nants_telescope >= Nants_data (which should always be true).

File: hera_cal/omni.py (hunk 1 above)
Comment (zakiali) [reply hera_cal/omni.py]: But you're looping over the antenna array which has length max(antenna_number). Or do antennas not in the array have antenna positions far from the size of the earth?

Comment (plaplant) [reply hera_cal/omni.py]: Yes, antennas not in the array are not "far" from the center of the Earth in absolute coordinates (typically a few meters, based on the default value of (-1, -1, -1) that then gets rotated to a different frame). So we loop over all antennas in the AntennaArray, but only real antennas will get added to the `ants_telescope` array, with a corresponding entry in `antpos`. The length of `ants_telescope` defines `Nants_telescope`, so right now we get 47.

**Pull request merged**